### ISSUE TITLE
Callback required when calling contract.setCandidate

### ISSUE THREAD

**Issue by olivdb** [dappuniversity/web3_tutorial#1]
When serving index.html from an HTTP Server, MetaMask complains that it requires a callback function when trying to set the candidate's name with ` contract.setCandidate($('input').val())`:

> Uncaught Error: The MetaMask Web3 object does not support synchronous methods like eth_sendTransaction without a callback parameter`

Adding a callback solved the issue:
```javascript
$('form').on('submit', function(event) {
    event.preventDefault();
    contract.setCandidate($('input').val(), function (err, result) {
        $('#candidateName').html($('input').val());
    });
});
```

**Comment by Manoj13-coder** [comment]
yes i recently check this dapp in my browser and i have to refresh my page again and again in order to update the candidate name . And i do a lot of things to make this  happen without refreshing the page but then i found a solution for this problem that what if we add location.reload() ; function in it so that whenever we click add candidate button it can automatically refresh the page in order to update it.

**Comment by Manoj13-coder** [comment]
it works fine

**Comment by Manoj13-coder** [comment]
yes i recently check this dapp in my browser and i have to refresh my page again and again in order to update the candidate name . And i do a lot of things to make this happen without refreshing the page but then i found a solution for this problem that what if we add location.reload() ; function in it so that whenever we click add candidate button it can automatically refresh the page in order to update it.

$('form').on('submit', function(event) {
        event.preventDefault();
		location.reload();
        contract.setCandidate($('input').val());
      })

**Comment by Domche-Michael** [comment]
Ok!
